### PR TITLE
Fix provider inputs serialization

### PR DESCRIPTION
**Pull request by pawelprazak**

### CODE REVIEW

File: core/src/main/scala/besom/internal/ProtobufUtil.scala [hunk 1]
```diff
@@ -7,7 +7,9 @@ import scala.util.*
 object ProtobufUtil:
   private val printer = JsonFormat.printer().omittingInsignificantWhitespace()
 
-  extension (v: Value) def asJsonString: Either[Throwable, String] = Try(printer.print(Value.toJavaProto(v))).toEither
+  extension (v: Value)
+    def asJsonString: Either[Throwable, String] = Try(printer.print(Value.toJavaProto(v))).toEither
+    def asJsonEncoded: Try[Value]               = v.asJsonString.map(_.asValue).toTry
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/ProtobufUtil.scala:12]: this seems to be dead code in the end, remove pls

File: core/src/test/scala/besom/internal/ArgsEncoderTest.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        suppressHelmHookWarnings = suppressHelmHookWarnings.asOptionOutput(isSecret = false)
+      )
+end ArgsEncoderTest
+
+class ArgsEncoderTest extends munit.FunSuite with ValueAssertions:
+  import ArgsEncoderTest.*
+
+  test("encode ProviderArgs") {
+    val pae       = summon[ProviderArgsEncoder[ProviderArgs]]
+    given Context = DummyContext().unsafeRunSync()
```
Comment (lbialy) [inline core/src/test/scala/besom/internal/ArgsEncoderTest.scala:57]: pardon, my bad, this context is unused here, right? can it be deleted?

File: core/src/test/scala/besom/internal/PropertiesSerializer.test.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package besom.internal
```
Comment (lbialy) [inline core/src/test/scala/besom/internal/PropertiesSerializer.test.scala:1]: name of this file is redundant, no need for .test.scala, it's in directory test

**Review by lbialy — CHANGES_REQUESTED** [review]
all of the provider/argsencoder tests are spluttered in different files, could you move them to a single file or just place in respective test suites, currently they are in:
* ContextTest (probably legacy naming, it doesn't seem to test context and there's not much to test there anyway)
* ArgsEncoderTest
* ProviderArgsEncoderTest

File: core/src/test/scala/besom/internal/ArgsEncoderTest.scala (hunk 2 above)
Comment (pawelprazak) [reply core/src/test/scala/besom/internal/ArgsEncoderTest.scala:57]: it is used, I've checked

**Pull request merged**